public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -32,14 +34,33 @@ export class PrometheusDatasource {
     this.basicAuth = instanceSettings.basicAuth;
     this.withCredentials = instanceSettings.withCredentials;
     this.interval = instanceSettings.jsonData.timeInterval || '15s';
+    this.httpMethod = instanceSettings.jsonData.httpMethod;
```

Comment:
This fails for existing datasources, because `httpMethod` is undefined till you open DS config and save it with a new option. I fixed this in #11087.

Reply:
Oops, sorry for my bug, and thanks for your fix.